src/dataframe/dataframe.jl

```diff
@@ -700,8 +697,8 @@ julia> insert!(d, 1, 'a':'c', :b)
 ```
 
 """
-function Base.insert!(df::DataFrame, col_ind::Int, item::AbstractVector, name::Symbol;
-                      makeunique::Bool=false)
+function insertcol!(df::DataFrame, col_ind::Int, item::AbstractVector, name::Symbol;
+                    makeunique::Bool=false)
```

Comment:
When updating this you might want to fix the following bug:
```
julia> df = DataFrame(a=Int[], b=[], c=[])
0×3 DataFrame


julia> insert!(df, 2, [1,2,3], :d)
0×4 DataFrame


julia> df
0×4 DataFrame


julia> df[:a]
0-element Array{Int64,1}

julia> df[:d]
3-element Array{Int64,1}:
 1
 2
 3
```

Reply:
Funny. I don't understand why the code explicitly allows a non-matching length when `size(df, 1) == 0`. Any idea?

Reply:
My idea was that it is a bug 😄. I assume the intention was `size(df, 2) == 0` so that we allow to add a column to a `DataFrame` that has `0` columns.

Actually that is why I prefer `nrows` and `ncols` to `size` (although I know that there are comments that those should be removed).

Reply:
Sounds plausible. I've pushed a commit to this that.